logger/drain/factory.go

```diff
@@ -0,0 +1,33 @@
+package drain
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/deis/deis/logger/drain/tcp"
+	"github.com/deis/deis/logger/drain/udp"
+)
+
+// NewDrain returns a pointer to an appropriate implementation of the LogDrain interface, as
+// determined by the drainURL it is passed.
+func NewDrain(drainURL string) (LogDrain, error) {
+	if drainURL == "" {
+		// nil means no drain-- which is valid
+		return nil, nil
+	}
+	if strings.HasPrefix(drainURL, "udp://") || strings.HasPrefix(drainURL, "syslog://") {
+		drain, err := udp.NewDrain(drainURL)
+		if err != nil {
+			return nil, err
+		}
+		return drain, nil
+	}
+	if strings.HasPrefix(drainURL, "tcp://") {
```

Comment:
How about `else if`? They're related so no need to execute this test if the first one match.